interpreter/exec/simd.ml

```diff
@@ -322,10 +338,43 @@ struct
       let num_lanes = lanes F64x2
     end)
 
+  let clamp low high x = min (max x low) high
+
+  (* Narrow two v128 into one v128 by using to_shape on both operands,
+   * concatenating them, clamping the element to between low and high,
+   * then of_shape to reconstruct a v128. *)
+  let narrow to_shape of_shape low high x y =
+    let xy = (to_shape x) @ (to_shape y) in
+    of_shape (List.map (clamp low high) xy)
+
+  module I8x16_convert = struct
+    let narrow_s = narrow Rep.to_i16x8 Rep.of_i8x16 (-128l) 127l
+    let narrow_u = narrow Rep.to_i16x8 Rep.of_i8x16 0l 255l
+  end
+
+  module I16x8_convert = struct
+    let narrow_s = narrow Rep.to_i32x4 Rep.of_i16x8 (-32768l) 32767l
+    let narrow_u = narrow Rep.to_i32x4 Rep.of_i16x8 0l 65535l
+
+    let widen take_or_drop mask x =
+      Rep.of_i16x8 (List.map (Int32.logand mask) (take_or_drop 8 (Rep.to_i8x16 x)))
+    let widen_low_s = widen Lib.List.take 0xffffffffl
+    let widen_high_s = widen Lib.List.drop 0xffffffffl
+    let widen_low_u = widen Lib.List.take 0xffl
+    let widen_high_u = widen Lib.List.drop 0xffl
+  end
+
   module I32x4_convert = struct
     let convert_using f v = Rep.of_i32x4 (List.map f (Rep.to_f32x4 v))
```

Comment:
Nit: I'd remove the `_using` suffix on this one, since you don't have it anywhere else.